consensus/src/alerts/mod.rs

```diff
@@ -490,3 +490,367 @@ pub(crate) async fn run<H: Hasher, D: Data, MK: MultiKeychain>(
+            let node_id = NodeIndex(i);
+            assert_eq!(
+                own_node
+                    .on_message(AlertMessage::AlertRequest(node_id, alert_hash))
+                    .await,
+                Some(AlerterResponse::ForkAlert(
+                    signed_alert.clone(),
+                    Recipient::Node(node_id),
+                )),
+            );
+        }
+    }
+
+    #[tokio::test]
+    async fn notifies_only_about_multisigned_alert() {
+        let n_members = NodeCount(7);
+        let own_index = NodeIndex(0);
+        let double_committer = NodeIndex(5);
+        let forker_index = NodeIndex(6);
+        let key_boxes: Vec<_> = (0..n_members.0)
+            .map(|i| KeyBox::new(n_members, NodeIndex(i)))
+            .collect();
+        let mut own_node = Alerter::new(
+            &key_boxes[own_index.0],
+            AlertConfig {
+                n_members,
+                session_id: 0,
+            },
+        );
+        let fork_proof =
+            make_fork_proof(forker_index, &key_boxes[forker_index.0], 0, n_members).await;
+        let empty_alert = Alert::new(double_committer, fork_proof.clone(), vec![]);
+        let empty_alert_hash = Signable::hash(&empty_alert);
+        let signed_empty_alert = Signed::sign(empty_alert.clone(), &key_boxes[double_committer.0])
+            .await
+            .into_unchecked();
+        assert_eq!(
+            own_node
+                .on_message(AlertMessage::ForkAlert(signed_empty_alert))
+                .await,
+            Some(AlerterResponse::ForkResponse(
+                Some(ForkingNotification::Forker(fork_proof.clone())),
+                empty_alert_hash,
+            )),
+        );
+        let forker_unit = fork_proof.0.clone();
+        let nonempty_alert = Alert::new(
+            double_committer,
+            fork_proof.clone(),
+            vec![forker_unit.clone()],
+        );
+        let nonempty_alert_hash = Signable::hash(&nonempty_alert);
+        let signed_nonempty_alert =
+            Signed::sign(nonempty_alert.clone(), &key_boxes[double_committer.0])
+                .await
+                .into_unchecked();
+        let signed_nonempty_alert_hash =
+            Signed::sign_with_index(nonempty_alert_hash, &key_boxes[double_committer.0])
+                .await
+                .into_unchecked();
+        let mut multisigned_nonempty_alert_hash = signed_nonempty_alert_hash
+            .check(&key_boxes[double_committer.0])
+            .expect("the signature is correct")
+            .into_partially_multisigned(&key_boxes[double_committer.0]);
+        for i in 1..n_members.0 - 2 {
+            let node_id = NodeIndex(i);
+            let signed_nonempty_alert_hash =
+                Signed::sign_with_index(nonempty_alert_hash, &key_boxes[node_id.0])
+                    .await
+                    .into_unchecked();
+            multisigned_nonempty_alert_hash = multisigned_nonempty_alert_hash.add_signature(
```

Comment:
Something went really wrong with this test, you meticulously create this multisignature and then don't use it in any way in the test.

I think you need to use this multisignature to create an RMC message and check whether that correctly responds with a `AlerterResponse::RmcMessage`. You should probably also check that a not fully multisigned RMC message of this nonempty alert gets ignored, and a not fully multisigned RMC message of the empty alert does not get ignored (at least before the alerter receives the multisigned version, it could theoretically ignore it afterwards although the current implementation doesn't).

Reply:
Yes, I feel like @artemis-cc  tried to do two different things at once in this test.

If I understand correctly this was supposed to be a test case where we learn about two alerts `a1`, `a2`, with the same `alert_id` but the second one `a2` (the one we received later) then gets multisigned. In such a case, upon getting a `complete` rmc message for `a2` we should output `AlerterResponse::RmcMessage(message)`. There seems to be a few lines missing then if this was the intention.